src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs

```diff
@@ -165,34 +169,34 @@ private void RegisterTocMap(TocItemViewModel item, string key, IDocumentBuildCon
             }
         }
 
-        private string ResolveHref(string originalPathToFile, FileModel model, IDocumentBuildContext context)
+        private string ResolveHref(string pathToFile, string originalPathToFile, FileModel model, IDocumentBuildContext context)
         {
-            if (!Utility.IsSupportedRelativeHref(originalPathToFile))
+            if (!Utility.IsSupportedRelativeHref(pathToFile))
             {
-                return originalPathToFile;
+                return pathToFile;
             }
 
-            var index = originalPathToFile.IndexOf('#');
+            var index = pathToFile.IndexOf('#');
             if (index == 0)
             {
-                throw new DocumentException($"Invalid toc link: {originalPathToFile}.");
+                throw new DocumentException($"Invalid toc link: {pathToFile}.");
             }
             string href = index == -1
-                ? context.GetFilePath(originalPathToFile)
-                : context.GetFilePath(originalPathToFile.Remove(index));
+                ? context.GetFilePath(pathToFile)
+                : context.GetFilePath(pathToFile.Remove(index));
 
 
             if (href == null)
             {
-                Logger.LogWarning($"Unable to find file \"{originalPathToFile}\" referenced by TOC file \"{model.LocalPathFromRepoRoot}\"");
+                Logger.LogWarning($"Unable to find file \"{pathToFile}\" referenced by TOC file \"{model.LocalPathFromRepoRoot}\"");
```

Comment:
>pathToFile [](start = 59, length = 10)

So does this, then user could see the original content he write in the toc file. But not see ~/ any more.